internal/bminventory/inventory.go

```diff
@@ -2372,7 +2384,8 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 		ingressVip = *params.ClusterUpdateParams.IngressVip
 	}
 	if params.ClusterUpdateParams.MachineNetworks != nil &&
-		common.IsSliceNonEmpty(params.ClusterUpdateParams.MachineNetworks) {
+		common.IsSliceNonEmpty(params.ClusterUpdateParams.MachineNetworks) &&
+		!reqDualStack {
```

Comment:
Do you think we can allow this for single stack too? I think it would be useful for the remote workers use-case to be able to set the VIP and API but also be able to set the `machineNetworks` were machines could exist. 

Not having a way to specify the worker's network will make the deployment fail `failure verifying CIDR of API VIP: Failed to calculate network CIDR: No suitable matching CIDR found for API VIP`